drivers/video/video_stm32_venc.c

```diff
@@ -16,8 +16,14 @@
 #include <zephyr/irq.h>
 #include <zephyr/kernel.h>
 #include <zephyr/logging/log.h>
+#if defined(CONFIG_VIDEO_STM32_VENC_SMH_ALLOC)
 #include <zephyr/multi_heap/shared_multi_heap.h>
 
+#define EWL_HEAP_ALIGNED_ALLOC(size)\
+	shared_multi_heap_aligned_alloc(CONFIG_VIDEO_BUFFER_SMH_ATTRIBUTE, ALIGNMENT_INCR, (size))
+#define EWL_HEAP_ALIGNED_FREE(block) shared_multi_heap_free(block)
+#endif
```

Comment:
Is mixing header files inclusion and macro definition really nice? Always including **shared_multi_heap.h** should not be an issue. Suggestion
```patch
 #include <zephyr/logging/log.h>
 #include <zephyr/multi_heap/shared_multi_heap.h>

 #include <ewl.h>
 
 (...)

 #define ALIGNMENT_INCR 8UL

+#if defined(CONFIG_VIDEO_STM32_VENC_SMH_ALLOC)
 #define EWL_HEAP_ALIGNED_ALLOC(size)\
 	shared_multi_heap_aligned_alloc(CONFIG_VIDEO_BUFFER_SMH_ATTRIBUTE, ALIGNMENT_INCR, (size))
 #define EWL_HEAP_ALIGNED_FREE(block) shared_multi_heap_free(block)
+#endif /* CONFIG_VIDEO_STM32_VENC_SMH_ALLOC */